src/script/sign.cpp

```diff
@@ -146,6 +146,16 @@ static bool CreateSig(const BaseSignatureCreator& creator, SignatureData& sigdat
 
 static bool CreateTaprootScriptSig(const BaseSignatureCreator& creator, SignatureData& sigdata, const SigningProvider& provider, std::vector<unsigned char>& sig_out, const XOnlyPubKey& pubkey, const uint256& leaf_hash, SigVersion sigversion)
 {
+    KeyOriginInfo info;
+    if (provider.GetKeyOriginByXOnly(pubkey, info)) {
+        auto it = sigdata.taproot_misc_pubkeys.find(pubkey);
+        if (it == sigdata.taproot_misc_pubkeys.end()) {
+            sigdata.taproot_misc_pubkeys.emplace(pubkey, std::make_pair(std::set<uint256>({leaf_hash}), info));
```

Comment:
not related to the PR, but I'm confused
`taproot_misc_pubkeys` will be serialised into `PSBT_IN_TAP_BIP32_DERIVATION`, but according to bip-0174

> Keys within each scope should never be duplicated

Can you help me understand how it works?

Reply:
`PSBT_IN_TAP_BIP32_DERIVATION` fields have keys that are composed of the type and the pubkey. As each pubkey is different, each key of this type will be different as well so there is no duplication. If a key appears in multiple tapscripts, we take the `else` path which only updates the set of leaf hashes.